atc/api/present/build.go

```diff
@@ -22,6 +22,7 @@ func Build(build db.Build) atc.Build {
 		ID:                   build.ID(),
 		Name:                 build.Name(),
 		JobName:              build.JobName(),
+		Comment:              build.Comment(),
```

Comment:
I think it's best to exclude the comment from the API response if the viewer is authorized to get the build but isn't authorized to see any build details, rather than just hiding it from the UI.

The easiest way would be for there to be a `GET /builds/.../comment` endpoint (rather than adding the comment to the build directly), but this makes it impractical to have a tooltip for build comments in the history bar, which I think is a pretty useful feature.

One imperfect way of accomplishing this: changing the signature of `present.Build` to `(build db.Build, showComment bool)`.

Most endpoints would just set `showComment` to `false`, since we don't care about build comments for the most part. However, `buildserver/get.go`, `jobserver/get_build.go`, and `jobserver/list_builds.go` should set `showComment` to `true` *if* the following holds:

```
showComments := accessor.GetAccessor(r).IsAuthorized(job.TeamName()) || job.Public()
```

(note: for `buildserver/get.go`, we'll have for existence of the job as well - if the build isn't for a job, or the job isn't public, then `showComments` should be `false`)

It's a bit awkward sprinkling this logic throughout - maybe there's a nicer way to abstract it, but this is the first thing that came to mind.

Reply:
Instead of passing in a `showComments` and leaving it up to the caller to decide when to include comments in the build response, I thought it might be cleaner to pass in the job the build is for and an instance of `accessor.Access`, so `present.Build` could make the decision.

https://github.com/concourse/concourse/blob/00c44311db661129331621224139be9d6398f600/atc/api/present/build.go#L12

https://github.com/concourse/concourse/blob/00c44311db661129331621224139be9d6398f600/atc/api/present/build.go#L36-L46

If the caller cannot provide the job or `accessor.Access`, then the comments are not included. This way the logic for deciding when to show comments does not need to be duplicated across all the different call sites.

Also added `db/build.Job` to (try to) retrieve `Job` from ` Build`, to make the `buildserver/get.go` case simpler.

https://github.com/concourse/concourse/blob/00c44311db661129331621224139be9d6398f600/atc/db/build.go#L424-L447

Reply:
This definitely seems like a nicer solution than what I suggested 👍